assets/src/stories-editor/components/higher-order/with-context-menu.js

```diff
@@ -47,9 +46,14 @@ const applyWithSelect = withSelect( ( select ) => {
 			editLayout.appendChild( menuWrapper );
 		}
 
+		// Make sure that user did not right click topbar.
+		const toolBar = document.querySelector( '.edit-post-header' );
+		if ( toolBar.compareDocumentPosition( event.target ) === 20 ) {
```

Comment:
I get why the value 20 is returned, but checking directly for that is irrelevant. Of course any node contained in another node follows it.

I still think the proper check is to do:

```suggestion
		if ( toolBar.compareDocumentPosition( event.target ) & Node.DOCUMENT_POSITION_CONTAINED_BY ) {
```

If you really want to validate, that it's both contained and followed by (not that the fact that it's followed by matters here) you should still use the bit mask pattern:

```suggestion
		const mask = Node.DOCUMENT_POSITION_CONTAINED_BY | Node. DOCUMENT_POSITION_FOLLOWING;
		if ( ( toolBar.compareDocumentPosition( event.target ) & mask ) === mask ) {
```

And not just plain integers. 

However the only important fact here is the `DOCUMENT_POSITION_CONTAINED_BY`, so no need to check for other than that in my opinion.

You could move it to a variable and wrap it in `Boolean()` for increased clarity:

```suggestion
		const isEventInsideToolbar = Boolean( toolBar.compareDocumentPosition( event.target ) & Node.DOCUMENT_POSITION_CONTAINED_BY );
		if ( isEventInsideToolbar ) {
```

Reply:
```

const isEventInsideToolbar = Boolean( toolBar.compareDocumentPosition( event.target ) & Node.DOCUMENT_POSITION_CONTAINED_BY );
--
  | if ( isEventInsideToolbar ) {

```
I get these errors


51:41  error  Unexpected use of '&'  no-bitwise
  51:91  error  'Node' is not defined  no-undef

Reply:
That rule should clearly be disabled for a use-case like this.

Create [a helper function akin to this one](https://github.com/StephenGrider/MeteorCasts/blob/master/markbin/node_modules/fbjs/lib/containsNode.js.flow), it contains the eslint disable as well. [It's pretty common for this use-case - even used in React core](https://www.codota.com/code/javascript/functions/builtins/containsNode).

I can live with 16 being directly used in the code, though I feel it's a bit of [a magic number](https://stackoverflow.com/questions/47882/what-is-a-magic-number-and-why-is-it-bad). `window.Node.DOCUMENT_POSITION_CONTAINED_BY` would probably be the correct notation.

Reply:
I can live with that too! I don't know if there might be issues with text nodes being event targets (the linked helper function has specific workarounds for those), but if it works, it works.

Reply:
Changed in https://github.com/ampproject/amp-wp/pull/3661/commits/d85df416c880a8b63b20ad84e459c1f23a9fa1f5